src/adapter/threads.ts

```diff
@@ -572,18 +572,41 @@ export class Thread implements VariableStoreDelegate {
     };
   }
 
-  rawLocationToUiLocation(rawLocation: RawLocation): Promise<PreferredUILocation | undefined> {
-    const script = rawLocation.scriptId ? this._scripts.get(rawLocation.scriptId) : undefined;
-    if (!script)
-      return Promise.resolve(undefined);
-    let {lineNumber, columnNumber} = rawToUiOffset(rawLocation, this.defaultScriptOffset());
+  public async rawLocationToUiLocation(rawLocation: RawLocation): Promise<PreferredUILocation | undefined> {
+    const script = rawLocation.scriptId ? await this.getScriptByIdOrWait(rawLocation.scriptId) : undefined;
+    if (!script) {
+      return;
+    }
+
+    const { lineNumber, columnNumber } = rawToUiOffset(rawLocation, this.defaultScriptOffset());
     return this._sourceContainer.preferredUiLocation({
       lineNumber,
       columnNumber,
       source: script.source
     });
   }
 
+  /**
+   * Gets a script ID if it exists, or waits to up maxTime. In rare cases we
+   * can get a request (like a stacktrace request) from DAP before Chrome
+   * finishes passing its sources over. We *should* normally know about all
```

Comment:
So you get a Paused event from Chrome that includes a script that you haven't gotten a scriptParsed event for yet? We should file a bug for that.

Reply:
It's actually

 1. We attach, initial events from CDP start coming, including script parsed events
 2. We hit a breakpoint and tell DAP
 3. DAP asks for the stacktrace, before we got `scriptParsed` events for all scripts in the stacktraces
 4. We respond and omit the `source` of scripts we haven't seen yet in stackframes.

This change delays step 4

Reply:
Right, but it seems like CDP shouldn't tell us that we hit a breakpoint until it tells us about the loaded scripts in the stack.